### PR TITLE
Promisified the youtube process.

### PR DESCRIPTION
**Pull request by ehgoodenough**
Because the process of retrieving a new video is going to include a lot of asynchronous steps (with lots and lots of callbacks after callbacks after callbacks), I used a library to wrap ytdl with promises. At the moment, it isn't affording us much, but it does formalize how we want to do what and when.

Also, I deleted the .jsrsrc, since we weren't using it.

### CODE REVIEW

**Comment by ehgoodenough** [comment]
This night class is so so boring.

selfie-0

File: source/youtube.process.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                resolve(file);
+            });
+            
+            process.pipe(fs.createWriteStream(file));
+        });
+    }
+}
+
+console.log(Date.now(), "Beginning Youtube Download");
+module.exports.download("FD_SbJCQovU").then(function()
```
Comment (tmjz402) [inline source/youtube.process.js]: Looks like some leftovers from testing?

Comment (ehgoodenough) [inline source/youtube.process.js]: Yup, definitely meant to remove that...

**Comment by ehgoodenough** [comment]
@tzorca, We're still trying to fix the conflicts and issues with Travis, but other than that, It'd be great if you could code review this and maybe merge me if we're all cool.

**Pull request merged**